src/components/common/Extensions/UserHub/partials/BalanceTab/partials/StreamsInfoRow/hooks.ts

```diff
@@ -3,51 +3,74 @@ import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useAppContext } from '~context/AppContext/AppContext.ts';
 import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
 import { useCurrencyContext } from '~context/CurrencyContext/CurrencyContext.ts';
-import { useGetStreamingPaymentsByColonyQuery } from '~gql';
+import { useGetStreamingPaymentsByColonyLazyQuery } from '~gql';
 import useCurrentBlockTime from '~hooks/useCurrentBlockTime.ts';
+import { StreamingPaymentStatus } from '~types/streamingPayments.ts';
 import { notNull } from '~utils/arrays/index.ts';
+import {
+  getStreamingPaymentAmountsLeft,
+  getStreamingPaymentStatus,
+} from '~utils/streamingPayments.ts';
 
 import { type StreamingPaymentItems } from './types.ts';
-import { calculateTotalsFromStreams } from './utils.ts';
+import {
+  calculateAverageStreamingPayment,
+  calculateTotalsFromStreams,
+} from './utils.ts';
 
-export const useStreamingPaymentsTotalFunds = () => {
+interface useStreamingPaymentsTotalFundsProps {
+  getDataByRecipentAddress?: boolean;
```

Comment:
I find this a bit confusing. Could `walletAddress` become a prop just like `nativeDomainId`?